### PR TITLE
Update the case test_backuptarget_available_during_engine_image_not_ready

### PR DESCRIPTION
**Pull request by roger-ryao**
Update the case test_backuptarget_available_during_engine_image_not_ready

issue https://github.com/longhorn/longhorn/issues/4750

Signed-off-by: Roger Yao <roger.yao@suse.com>

### CODE REVIEW

File: manager/integration/tests/test_basic.py [hunk 1]
```diff
@@ -3769,8 +3769,7 @@ def test_workload_with_fsgroup(core_api, statefulset):  # NOQA
 # After introducing the gRPC proxy, the backup target controller is relying on
 # the gRPC server availability instead of the engine binary availability.
 # https://github.com/longhorn/longhorn/issues/4033
```
[the remark this answers is not in the document]
Comment (roger-ryao) [reply manager/integration/tests/test_basic.py]: Done

**Review by c3y1huang — APPROVED** [review]

**Comment by roger-ryao** [comment]
Update [PR-1178](https://github.com/longhorn/longhorn-tests/pull/1178) from @innobead's suggestions
- migrate cleanup function to common.py
  - [regression test report](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2469/)

File: manager/integration/tests/common.py [hunk 2]
```diff
@@ -3125,6 +3126,25 @@ def reset_engine_image(client):
...
 
+# ensure that the engine image daemonset scale up for longhorn
+# after scaling down daemonset
+def scale_up_engine_image_daemonset(client):
+    apps_api = get_apps_api_client()
+    default_img = get_default_engine_image(client)
+    ds_name = "engine-image-" + default_img.name
+    body = [{
+        "op": "remove",
+        "path": "/spec/template/spec/nodeSelector/foo"
```
Comment (c3y1huang) [inline manager/integration/tests/common.py:3137]: So this only removed the node selector `foo`. Is it possible to remove all node selectors?

**Comment by roger-ryao** [comment]
Update [PR-1178](https://github.com/longhorn/longhorn-tests/pull/1178) from @c3y1huang's suggestions
- updated cleanup function and remove all node selectors
  - [the regression test report](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2501/testReport/)
  - [the test case test report](https://ci.longhorn.io/job/private/job/longhorn-tests-regression/2497/)

**Review by c3y1huang — APPROVED** [review]

**Comment by roger-ryao** [comment]
> Please cherry-pick this to #1180

Done

**Pull request merged**